src/ansiblelint/transformer.py

```diff
@@ -0,0 +1,164 @@
+"""Transformer implementation."""
+import logging
+import re
+from typing import Dict, List, Set, Union, cast
+
+from ruamel.yaml.comments import CommentedMap, CommentedSeq
+from ruamel.yaml.emitter import Emitter
+
+# Module 'ruamel.yaml' does not explicitly export attribute 'YAML'; implicit reexport disabled
+# To make the type checkers happy, we import from ruamel.yaml.main instead.
+from ruamel.yaml.main import YAML
+
+from .errors import MatchError
+from .file_utils import Lintable
+from .runner import LintResult
+from .skip_utils import load_data  # TODO: move load_data out of skip_utils
+
+__all__ = ["Transformer"]
+
+_logger = logging.getLogger(__name__)
+
+# ruamel.yaml only preserves empty (no whitespace) blank lines
+# (ie "/n/n" becomes "/n/n" but "/n  /n" becomes "/n").
+# So, we need to identify whitespace-only lines to drop spaces before reading.
+_whitespace_only_lines_re = re.compile(r"^ +$", re.MULTILINE)
+
+
+class FormattedEmitter(Emitter):
+    """Emitter that applies custom formatting rules when dumping YAML.
+
+    Root-level sequences are never indented.
+    All subsequent levels are indented as configured (normal ruamel.yaml behavior).
+
+    Earlier implementations used dedent on ruamel.yaml's dumped output,
+    but string magic like that had a ton of problematic edge cases.
+    """
+
+    _sequence_indent = 2
+    _sequence_dash_offset = 0  # Should be _sequence_indent - 2
+
+    # NB: mypy does not support overriding attributes with properties yet:
+    #     https://github.com/python/mypy/issues/4125
+    #     To silence we have to ignore[override] both the @property and the method.
+
+    @property  # type: ignore[override]
+    def best_sequence_indent(self) -> int:  # type: ignore[override]
+        """Return the configured sequence_indent or 2 for root level."""
+        return 2 if self.column < 2 else self._sequence_indent
```

Comment:
I'm not sure I know what `self.column < 2` means. I think it's determining whether we're at the root? We might want to add a comment or extract a function for that.

Reply:
done